### ISSUE TITLE
Fix framework pack NuGet package licenseUrl

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#8694]
For example these packages link to the license via a "floating" Core-Setup GitHub repo URL:
[Microsoft.NETCore.App 2.1.13](https://www.nuget.org/packages/Microsoft.NETCore.App/2.1.13)
[Microsoft.NETCore.App.Ref 3.0.0](https://www.nuget.org/packages/Microsoft.NETCore.App.Ref/3.0.0)
[Microsoft.NETCore.App.Ref 3.1.0-preview1.19506.1](https://www.nuget.org/packages/Microsoft.NETCore.App.Ref/3.1.0-preview1.19506.1)
They link to https://github.com/dotnet/core-setup/blob/master/LICENSE.TXT based on `licenseUrl` in the nuspec.

This is awkward for repo consolidation (we have to create a probably confusing `master` tag to avoid breaking the link, rather than only having the `master-archive` tag). It's also very bad practice. NuGet packages should contain their license.

Mitigation: the NuGet packages *do* contain `LICENSE.TXT`, but the NuGet Gallery (and probably Visual Studio?) show `licenseUrl`, so the user would have to dig in their NuGet cache or extract the nupkg to find `LICENSE.TXT`.

Note that the managed library packages like [Microsoft.DotNet.PlatformAbstractions 3.0.0](https://www.nuget.org/packages/Microsoft.DotNet.PlatformAbstractions/3.0.0) aren't affected because they are packed using different packaging tooling.

The `pkgproj` packaging tooling should support what we need to do here, but something in Core-Setup is not right or interferes.

I believe we should definitely fix this for 3.1, and consider porting to 3.0.

/cc @MichaelSimons @dleeapho @leecow @ericstj

**Comment by dleeapho** [comment]
We should  fix for 5.0 and make 3.1 backport decision then.,

**Comment by ericstj** [comment]
https://github.com/dotnet/corefx/issues/42264

**Comment by ericstj** [comment]
Looks like the requirement is that the LicenseUrl property is *not* defined.

**Comment by dagood** [comment]
I tried removing the LicenseUrl override in Core-Setup and got packages with no `LicenseUrl` *and* no other license thing in the nuspec. They did contain `LICENSE.TXT` (as always) but my understanding is there should also be some `<license type="` in the nuspec linking it up or referencing a license by name.

**Comment by ericstj** [comment]
Is `$(PackageLicenseExpression)` set?

**Comment by dagood** [comment]
Oh--no, I expected an Arcade default. 😄 Set to `MIT`, and it works:

```
    <license type="expression">MIT</license>
    <licenseUrl>https://licenses.nuget.org/MIT</licenseUrl>
```

**Comment by tmat** [comment]
License URL intentionally does not have a default - each repo must make a decision what license to use. See https://github.com/dotnet/arcade/blob/master/Documentation/ArcadeSdk.md#licensetxt

**Comment by dagood** [comment]
Yeah, I think I had wires crossed from when I assumed CoreFX was working properly and didn't see it defining one. Might be nice for Arcade to produce an error when there's no license, but that might either be up to the CoreFX tooling or interfere with it, so no big deal IMO.

**Comment by tmat** [comment]
We do validate licenses, but it turns out only if a license expression is specified:
https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryValidation.proj#L14

That might have been temporary while not all repos had it. Feel free to follow up and make it require the expression to be specified.

**Comment by tmat** [comment]
We might have also relied on NuGet pack validating that packages have licenses. Not sure if it actually does.

**Comment by ericstj** [comment]
> CoreFX was working properly and didn't see it defining one.

We had to turn it off because of a NuGet bug on pack, but never turned it back on after working around that bug.
